### PR TITLE
fix(guardian): minimize PS for deployment

### PR DESCRIPTION
**Pull request by freecaykes**
## Description

minimize the PodSecurity of Guardian to fix deployment 

it was previously erroring out in kadm

```
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"

...

status:
  conditions:

  ....

  - lastTransitionTime: "2022-09-20T23:38:39Z"
    lastUpdateTime: "2022-09-20T23:38:39Z"
    message: 'pods "tigera-guardian-65f9b97c57-4lfp5" is forbidden: violates PodSecurity
      "restricted:latest": unrestricted capabilities (container "tigera-guardian"
      must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or
      container "tigera-guardian" must set securityContext.seccompProfile.type to
      "RuntimeDefault" or "Localhost")'
```

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/guardian.go
Comment (caseydavenport) [inline pkg/render/guardian.go:265]: "minimize PZ for deployment" isn't a super descriptive comment, and won't be useful to folks reading this down the line.

I would expect something like this:

"Build a security context for the pod that will allow the pod to be deployed. Guardian is run in a namespace with a pod security standard of X, which requires that the security context meet certain criteria in order for pods to be accepted by the API server".

**Review by caseydavenport — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
